errors.go

```diff
@@ -152,6 +152,34 @@ func ReplaceNativeMessage(e error, msg string) Error {
 	}
 }
 
+// AsType "casts" an input error as a target merry.Error such that the returned merry Error evaluates
+// to true when operating Is(newErr, targetErr). If e is nil, return nil. If targetType is nil,
+// A wrapped merry Error of e is returned. The key value pairs of the targetType is kept and then
+// appended with key value map from the input error. Stack trace of the original input error is maintained.
+func AsType(e error, targetType Error) (Error, bool) {
+	if targetType == nil {
+		return Wrap(e), false
+	}
+	switch e.(type) {
+	case nil:
+		return nil, false
+
+	case *merryErr:
+		output := targetType.Here()
+		output = output.WithValue(stack, Value(e, stack))
+
+		inputMap := Values(e)
+		for k, v := range inputMap {
+			output = output.WithValue(k, v)
+		}
+		return Append(output, e.Error()), true
```

Comment:
I'd like to propose an alternate approach; rather than reconstructing a whole new error chain and trying to copy all the relevant info over, what about adding explicit support for multiple identities to .Is?

In other words, implementation would be like:

```go
// This is the pattern used by userMessage, httpCode, etc. 
const additionalErrorIdentities = "additional error identities"

func (e *merryErr) WithErrorIdentity(err error) Error {
   var idents []error
    if i, ok := e.Value(additionalErrorIdentities).([]error) {
        idents = i
    }
    append(idents, err)
    return e.WithValue(additionalErrorIdentities, idents)
}
```

Then .Is would know how to check additionalErrorIdentities.